tests/GroupTest.php

```diff
@@ -181,71 +181,64 @@ public function testAddGroupSecondWay(): void
-                            }
-                        )
-                    );
-                        $group->addGroup(
-                        new Group(
-                            '/blog', static function (Group $group) {
-                                $group->addRoute(Route::get('/post')->name('api-v1-blog-post/index'));
-                                $group->addRoute(Route::get('/post/{id}')->name('api-v1-blog-post/view'));
-                            }
-                        )
-                    );
-                        $group->addRoute(Route::get('/note')->name('api-v1-note/index'));
-                        $group->addRoute(Route::get('/note/{id}')->name('api-v1-note/view'));
-                    }
-                )
-            );
-                $group->addGroup(
-                new Group(
-                    '/v2', static function (Group $group) {
-                        $group->addRoute(Route::get('/user')->name('api-v2-user/index'));
-                        $group->addRoute(Route::get('/user/{id}')->name('api-v2-user/view'));
-                        $group->addGroup(
-                        new Group(
-                            '/news', static function (Group $group) {
-                                $group->addRoute(Route::get('/post')->name('api-v2-news-post/index'));
-                                $group->addRoute(Route::get('/post/{id}')->name('api-v2-news-post/view'));
-                            }
-                        )
-                    );
-                        $group->addGroup(
-                        new Group(
-                            '/blog', static function (Group $group) {
-                                $group->addRoute(Route::get('/post')->name('api-v2-blog-post/index'));
-                                $group->addRoute(Route::get('/post/{id}')->name('api-v2-blog-post/view'));
-                            }
-                        )
-                    );
-                        $group->addRoute(Route::get('/note')->name('api-v2-note/index'));
-                        $group->addRoute(Route::get('/note/{id}')->name('api-v2-note/view'));
-                    }
-                )
-            );
-            }, $container
+        $routes = [
+            Route::get(''),
+            Route::post(''),
+            Route::delete(''),
+            Route::patch(''),
+            Route::put(''),
+            Route::head(''),
+            Route::options(''),
+            Route::anyMethod(''),
+        ];
+        $group = new Group(
+            '',
+            static function (Group $group) use ($routes) {
+                array_map( fn(Route $route) => $group->addRoute($route),$routes);
+            },
+            $this->getContainer()
         );
 
-        $items = $apiGroup->getItems();
+        array_map(fn(Route $route) => $this->assertFalse($route->hasContainer()), $routes);
```

Comment:
Would using data provider be more readable?

Reply:
Which one?